src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2Extensions.cs

```diff
@@ -203,5 +203,31 @@ public static ArraySegment<byte> GetArraySegment(this MemoryPoolIterator2 start,
             start.CopyTo(array, 0, length, out length);
             return new ArraySegment<byte>(array, 0, length);
         }
+
+        public static bool StartsWithIgnoreCase(this MemoryPoolIterator2 start, string prefix, out MemoryPoolIterator2 end, out bool caseMatches)
```

Comment:
You may want to profile this, we went through extreme lengths to avoid stuff like this on the per request path (in kestrel at least) :smile: